proto/info/info.proto

```diff
@@ -1068,8 +1082,37 @@ message ZInfoNetworkInstance {
   // on their guest side and cannot determine and report what their values are.
   // The MTU configured for the network instance by the user may differ from the actual MTU
   // used and reported here if the configured MTU is invalid or conflicts with the MTU
-  // of the associated device port (see NetworkInstanceConfig.mtu in netinst.proto).
+  // of the associated device port(s) (see NetworkInstanceConfig.mtu in netinst.proto).
   uint32 mtu = 42;
+
+  // Routes configured for the network instance.
+  // This includes connected routes, DHCP-received routes, user-defined static routes
+  // (NetworkInstanceConfig.static_routes) and the default route (if any).
+  // Note that some user-defined static routes might not be applied (and thus not reported
+  // here) if they do not match IP config of currently used device ports.
+  // Additionally, static routes with shared port labels (matching multiple ports)
+  // are reported here each with the logical label of the (single) port, currently selected
+  // for the route (selected based on connectivity status, port costs, wwan signal strength,
+  // etc.).
+  repeated IPRoute ip_routes = 43;
+}
+
+// Info about IP route configured for a network instance.
+message IPRoute {
+  // Destination network address in the CIDR format: <IP-address>/<prefix-length>
+  string destination_network = 1;
+  // Gateway IP address.
+  // Empty for connected route.
+  string gateway = 2;
+  // Logical label of the output device port for the routed traffic.
+  // Empty if the gateway is IP address of one of the applications. In that case,
+  // 'gateway_app' is defined instead.
+  string port = 3;
+  // UUID of the application used as the gateway for the route.
+  // Empty if the gateway is external (not one of the apps but outside of the device)
+  // or this is a connected route for a port (thus gateway is empty).
+  // In that case, 'port' is defined instead.
+  string gateway_app = 4;
```

Comment:
I'm reading the separate comments and it seems like they are saying that exactly one of gateway, port, and gateway_app is set. Or maybe gateway+port can both be set (useful if gateway is ever an IPv6 link-local).

It would be easier to understand the choices if there was a single comment explaining how the next hop can be specified instead of the per-field comments.

Reply:
Added comment about next-hop for `message IPRoute` for clarification:
```
// The next-hop for the IP route is specified by the "gateway" IP address
// (empty for a connected route) and the output "port". However, if the gateway
// is one of the applications running on the device, the port is actually a virtual
// interface with an EVE-generated interface name and no user-assigned logical label.
// Therefore, in such cases, it is more useful to leave "port" empty and instead use
// the "gateway_app" field to publish the UUID of the application used as the gateway.
```

Reply:
That helps. However, it doesn't say whether gateway can/should be empty when its is an application running on the device, plus a reader will still try to parse comments for the individual fields  and their head will hurt just like mine.

How about also simplifying the comments for the fields to e.g.,
// Optional logical label of the output device for the route. Can not be used when the gateway is an app instance of the device.
string port ...
// Optional UUID of the app instance used as the gateway for the route.
string gateway_app ...

Reply:
But this is IP route definition for the `ZInfoNetworkInstance` info message (I think it makes sense to publish all routes configured for NI). So this is about what fields to expect, why sometimes they are empty, etc. It is not about what the user should set.

Reply:
Ok